skore/src/skore/sklearn/_plot/metrics/roc_curve.py

```diff
@@ -240,32 +215,44 @@ def _plot_single_estimator(
             )
 
         else:  # multiclass-classification
+            labels = self.roc_curve["label"].unique()
             class_colors = sample_mpl_colormap(
-                colormaps.get_cmap("tab10"), 10 if len(self.fpr) < 10 else len(self.fpr)
+                colormaps.get_cmap("tab10"), 10 if len(labels) < 10 else len(labels)
             )
 
-            for class_idx, class_label in enumerate(self.fpr):
-                fpr_class = self.fpr[class_label][0]
-                tpr_class = self.tpr[class_label][0]
-                roc_auc_class = self.roc_auc[class_label][0]
+            for class_idx, class_label in enumerate(labels):
+                roc_curve = _filter_by(
+                    self.roc_curve,
+                    label=class_label,
+                )
```

Comment:
To illustrate what I meant by using `.query` directly:

```suggestion
                roc_curve = self.roc_curve.query("label == @class_label")
```

Reply:
done